### DISCUSSION TITLE
Kickoff Meeting

### DISCUSSION THREAD

**Discussion by 1cg** [bigskysoftware/missing#3] [Ideas]
# Kickoff Meeting for missing.css

Welcome!  We are in a position to have a kick off meeting for missing.css (the missing style sheet).

We have the following volunteers:

* @dz4k - Lead
* @1cg - Developer
* @benpate - Developer
*  @jvosloo - Ideas/Feedback

We will have the first meeting via zoom on Wednesday, Feb 16th at 10 MST/14 GMT:

https://www.worldtimebuddy.com/?pl=1&lid=0,745044,5641727&h=5641727&hf=0

## Zoom Info

https://us04web.zoom.us/j/73370779708?pwd=3g1H6f82LSFn2WsNylwcuAhVg3eKr0.1

## Tentative Agenda

* Introductions
* Carson - Purpose of missing.css
* Deniz - present basic ideas for syntax
* Carson - a plea for a depth variable
* A Review of classless.css project
* Decide on layout of missing.css project
    * Divide work up between developers (possibly delay pending exploratory work)
* Discuss how project will be built and distributed
    * https://missing.style - main website
    * https://the.missing.style - the latest version of the style sheet

**Comment by 1cg** [comment]
@fractaledmind @maco @carlotm let me know if you are interested in joining this zoom

**Reply by fractaledmind** [reply]
I think I can make it

**Comment by 1cg** [comment]
Recommended for tomorrows zoom, a close reading of classless.css's excellent style sheet:

https://github.com/emareg/classlesscss/blob/master/docs/classless.css

**Reply by fractaledmind** [reply]
One note, I think using `:where()` to manage complex selectors' specificity will help us (I think) to allow class-based utilities to override defaults

**Comment by waldyrious** [comment]
Just to be sure, are you guys aware of the list of classless (and class-light) stylesheets at https://github.com/dbohdan/classless-css?

**Reply by benpate** [reply]
This is a great list, and should absolutely inform what we're doing here.  Missing.css will have to add something interesting and better than what's already out there.  What *should* be the differentiator?

**Reply by fractaledmind** [reply]
I think the best differentiator is default styling of accessible and opinionated HTML structures. Adrian Roselli has a series of posts on his blog about "under engineered" UI widgets (e.g. https://adrianroselli.com/2021/12/under-engineered-dependency-questions.html). You can find excellent examples of CSS selectors that style accessibly structured HTML. Mix that with a foundation of utility classes and I think you have something powerful

**Reply by waldyrious** [reply]
> Missing.css will have to add something interesting and better than what's already out there. What _should_ be the differentiator?

Totally agreed. I checked the READMEs of all the classless styles and noticed none of them mention ARIA roles. So that sounds like it could be a differentiator of missing.css.

That said, I then searched around a bit and came across [Ariato](https://ariato.org). Quoting from the [README](https://github.com/BaseSecrete/ariato#ariatocss):

> _ariato.css_ first targets classless HTML elements.
>
> Then, _ariato.css_ targets classless Wai-aria roles and states to style other usual web application components.
>
> Only after that, if no existing html element or wai-aria role does the job, it uses minimal css classes. (ex: `.card`)

There definitely seems to be some overlap there. Quoting @1cg from [Discord](https://discord.com/channels/725789699527933952/941815579440992337/941833604797968434), in a statement I really liked and helped me get a clearer idea of what missing.css could be (emphasis mine):

> High level statement: A CSS library that starts with good default styling, akin to classless CSS libraries, that offers purely semantic HTML components, utilizing ARIA where appropriate and, finally, where tweaks are needed, offers a small & curated set of class-based utilities. 
> **We want users to have to go beyond bog-standard HTML as little as possible, but no littler.**

It will be important to validate what projects like Ariato are doing to match this vision, and what missing.css will do differently.

**Comment by benpate** [comment]
I'm very sorry, but I won't be able to make this meeting. I have another scheduled meeting every Wednesday at ten, and I can't miss this one.

I'll do my best to participate in the next one 😊
